website/addons/dryad/views.py

```diff
@@ -0,0 +1,225 @@
+    ret.update(dryad.update_json())
+    ret.update(_view_project(node, auth, primary=True))
+
+    return ret
+
+@must_be_valid_project
+@must_have_addon('dryad', 'node')
+def search_dryad_page(**kwargs):
+    node = kwargs['node']
+    pid = kwargs['pid']
+    auth = kwargs['auth']
+    count = 10
+    if "count" in request.args:
+        count = int(request.args["count"])
+    start = 0
+    if "start" in request.args:
+        start = int(request.args["start"])
+    query = ""
+    if "query" in request.args:
+        query = request.args["query"]
+
+    dryad = node.get_addon('dryad')
+
+    d = Dryad_SOLR()
+    x = d.basic_query(query)
+
+    docs = x.getElementsByTagName("doc")
+
+    count = int(x.getElementsByTagName("result")[0].attributes["numFound"].value)
+    start = int(x.getElementsByTagName("result")[0].attributes["start"].value)
+    total = int(x.getElementsByTagName("result")[0].attributes["numFound"].value)
+
+    ret = {"end": start + count,
+            "start": start,
+            "total": total,
+            "content": "",
+            'next_dryad': request.path + "?count={}&start={}".format(count, start + count),
+            'previous_dryad': request.path + "?count={}&start={}".format(count, start - count)}
+
+    objectList = ET.Element("ul")
+    for doc in docs:
+        title_elements = [i.firstChild.firstChild.wholeText for i in doc.getElementsByTagName("arr") if i.hasAttribute("name") and i.getAttribute("name") == "dc.title_ac"]
+        title_element = ""
+        if len(title_elements) > 0:
+            title_element = title_elements[0]
+
+        objInfo = ET.SubElement(objectList, "li")
+        objInfo.text = title_element
+        sublist = ET.SubElement(objInfo, "ul")
+
+        author_list = [i for i in doc.getElementsByTagName("arr") if i.hasAttribute("name") and i.getAttribute("name") == "dc.contributor.author_ac"][0]
+        author_list = [i.firstChild.wholeText for i in author_list.getElementsByTagName("str")]
+        authors = "Authors: " + ", ".join(author_list)
+
+        authorel = ET.SubElement(sublist, "li")
+        authorel.text = authors
+
+        identifier = [i.firstChild.firstChild.wholeText for i in doc.getElementsByTagName("arr") if i.hasAttribute("name") and i.getAttribute("name") == "dc.identifier"]
+        identifier = identifier[0]
+        id_item = ET.SubElement(sublist, "li")
+        id_button = ET.SubElement(id_item, "a")
+        id_button.text = "View Package Externally"
+        id_button.attrib["href"] = "http://datadryad.org/resource/{}".format(identifier) if "http" not in identifier else identifier
+
+        if "doi" in identifier:
+            add_item = ET.SubElement(sublist, "li")
+            add_button = ET.SubElement(add_item, "a")
+            add_button.text = "Set Node Data to This"
+            add_button.attrib["href"] = "/project/{}/dryad/add?doi={}".format(pid, identifier)
+
+    ret.update({"content": ET.tostring(objectList)})
```

Comment:
I would vasty prefer seeing HTML construction strictly limited to the template layer. Is there a reason this construction can't happen in a Mako template?